src/String.elm

```diff
@@ -13,7 +13,8 @@ module String exposing
   )
 
 {-| A built-in representation for efficient string manipulation. String literals
-are enclosed in `"double quotes"`. Strings are *not* lists of characters.
+are enclosed in `"double quotes"`. Strings are *not* lists of characters, but
+lists of UTF-16 code units.
```

Comment:
I think the existing wording ("Strings are *not* lists of characters.") was meant to clarify that a `String` is not `List` (unlike, e.g., [in Haskell](https://hackage.haskell.org/package/base-4.16.1.0/docs/Prelude.html#t:String), where `type String = [Char]`).

I think the reference to UTF-16 is fine (but maybe could be explained in terms of `Char`), but I don't think we should use the word "list". For example, the Revised^6 Report on the Algorithmic Language Scheme—another language where "list" has a very specific meaning—defines the character and string types like this:
> [**Characters**](http://www.r6rs.org/final/html/r6rs/r6rs-Z-H-2.html#node_toc_node_sec_Temp_8)
> Scheme characters mostly correspond to textual characters. More precisely, they are isomorphic to the *scalar values* of the Unicode standard.
> [**Strings**](http://www.r6rs.org/final/html/r6rs/r6rs-Z-H-2.html#node_toc_node_sec_Temp_9)
> Strings are finite sequences of characters with fixed length and thus represent arbitrary Unicode texts.